stream/src/test/java/com/annimon/stream/internal/SpinedBufferTest.java

```diff
@@ -243,85 +320,100 @@ public void testDoubleAccept() {
+        // test when special case - one chunk(<16 elements) present
+        b.get(10);
+    }
+
     @Test(expected = IndexOutOfBoundsException.class)
     public void testIntGetOutOfBounds() {
         SpinedBuffer.OfInt b = new SpinedBuffer.OfInt();
-
-        for(int i = 0; i < 5; i++)
+        for (int i = 0; i < 5; i++)
             b.accept(i);
-
         // test when special case - one chunk(<16 elements) present
         b.get(10);
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void testLongGetOutOfBounds() {
         SpinedBuffer.OfLong b = new SpinedBuffer.OfLong();
-
-        for(long i = 0; i < 5; i++)
+        for (long i = 0; i < 5; i++)
             b.accept(i);
-
         // test when special case - one chunk(<16 elements) present
         b.get(10);
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void testDoubleGetOutOfBounds() {
         SpinedBuffer.OfDouble b = new SpinedBuffer.OfDouble();
-
-        for(int i = 0; i < 5; i++)
+        for (int i = 0; i < 5; i++)
             b.accept(i);
-
         // test when special case - one chunk(<16 elements) present
         b.get(10);
     }
 
 
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testGetOutOfBounds2() {
+        SpinedBuffer.Of<String> b = new SpinedBuffer.Of<String>();
+        for (int i = 0; i < 32; i++)
+            b.accept("a");
+        // test when spine(several chunks, >16 elements)
+        b.get(40);
+    }
+
     @Test(expected = IndexOutOfBoundsException.class)
     public void testIntGetOutOfBounds2() {
         SpinedBuffer.OfInt b = new SpinedBuffer.OfInt();
-
-        for(int i = 0; i < 32; i++)
+        for (int i = 0; i < 32; i++)
             b.accept(i);
-
         // test when spine(several chunks, >16 elements)
         b.get(40);
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void testLongGetOutOfBounds2() {
         SpinedBuffer.OfLong b = new SpinedBuffer.OfLong();
-
-        for(long i = 0; i < 32; i++)
+        for (long i = 0; i < 32; i++)
             b.accept(i);
-
         // test when spine(several chunks, >16 elements)
         b.get(40);
     }
 
     @Test(expected = IndexOutOfBoundsException.class)
     public void testDoubleGetOutOfBounds2() {
         SpinedBuffer.OfDouble b = new SpinedBuffer.OfDouble();
-
-        for(int i = 0; i < 32; i++)
+        for (int i = 0; i < 32; i++)
             b.accept(i);
-
         // test when spine(several chunks, >16 elements)
         b.get(40);
     }
 
 
+    @SuppressWarnings("RedundantCast")
     @Test
-    public void testIntEnsureCapacity() {
-        SpinedBuffer.OfInt b = new SpinedBuffer.OfInt();
-
+    public void testEnsureCapacity() {
+        SpinedBuffer.Of<String> b = new SpinedBuffer.Of<String>();
         int m = SpinedBuffer.MIN_CHUNK_POWER;
+        int count = (1<<m) + (1<<m) + (1<<m+1) + (1<<m+2) + (1<<m + 3) + (1<<m+4) + (1<<m+5) + (1<<m+6);
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [BSHIFT: Possible bad parsing of shift operation (BSHIFT_WRONG_ADD_PRIORITY)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bshift-possible-bad-parsing-of-shift-operation-bshift-wrong-add-priority).
The code performs an operation like (x << 8 + y). Although this might be correct, probably it was meant to perform (x << 8) + y, but shift operation has a lower precedence, so it's actually parsed as x << (8 + y).

Reply:
`1 << (m+4)` was intended. For `MIN_CHUNK_POWER` = 4, `count` should be 2048.

Reply:
I see.

Reply:
I see.